plugin/plugin.go

```diff
@@ -21,27 +25,77 @@ func New(policy string) validator.Plugin {
 }
 
 func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
-
-	var droneConfig DroneConfig
-
-	err := yaml.Unmarshal([]byte(req.Config.Data), &droneConfig)
+	f := bytes.NewBufferString(req.Config.Data)
+	resources, err := Parse(f)
 	if err != nil {
 		return err
 	}
+	for _, resource := range resources {
 
-	r := rego.New(
-		rego.Query("deny = data.drone.validation.deny; msg = data.drone.validation.out"),
-		rego.Load([]string{p.policyPath}, nil),
-		rego.Input(droneConfig))
+		r := rego.New(
+			rego.Query("deny = data.drone.validation.deny; msg = data.drone.validation.out"),
+			rego.Load([]string{p.policyPath}, nil),
+			rego.Input(resource))
 
-	rs, err := r.Eval(ctx)
-	if err != nil {
-		return err
+		rs, err := r.Eval(ctx)
+		if err != nil {
+			return err
+		}
+		if rs[0].Bindings["deny"] == true {
+			message := fmt.Sprintf("pipeline %v", rs[0].Bindings["msg"])
+			return errors.New(message)
+		}
+
+	}
+	return nil
+}
+
+func Parse(r io.Reader) ([]*DroneConfig, error) {
+	const newline = '\n'
+	var resources []*DroneConfig
+	var resource *DroneConfig
+
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if isSeparator(line) {
+			resource = nil
+		}
+		if resource == nil {
+			resource = &DroneConfig{}
+			resources = append(resources, resource)
+		}
+		if isSeparator(line) {
+			continue
+		}
+		if isTerminator(line) {
+			break
```

Comment:
a terminator could exist in the middle of a file, so this seems to stop processing too early. Probably should add a case of this to unit tests.

However, see previous comment about using string splitting or yaml decoding instead.